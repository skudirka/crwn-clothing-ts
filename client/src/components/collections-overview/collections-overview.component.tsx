import React from 'react';
import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect';

import {selectCollectionsForPreview} from '../../redux/shop/shop.selectors';

import CollectionPreview from '../collection-preview/collection-preview.component';

import {CollectionsOverviewContainer} from './collections-overview.styles';

export const CollectionsOverview = ({collections}) => (
    <CollectionsOverviewContainer className="collections-overview">
        {
            collections.map(({id, ...otherCollectionProps}) => (
                <CollectionPreview key={id} {...otherCollectionProps} />
            ))
        }
    </CollectionsOverviewContainer>
);

const mapToStateProps = createStructuredSelector({
    collections: selectCollectionsForPreview
});

export default connect(mapToStateProps)(React.memo(CollectionsOverview));