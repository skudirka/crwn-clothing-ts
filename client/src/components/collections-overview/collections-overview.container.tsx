import {createStructuredSelector} from 'reselect';
import {connect, ConnectedProps} from 'react-redux';
import {compose} from 'redux';

import {selectIsCollectionFetching} from '../../redux/shop/shop.selectors';

import CollectionsOverview from './collections-overview.component';
import WithSpinner, {WithSpinnerProps} from '../with-spinner/with-spinner.component';

import {IRootState} from '../../redux/root-types';

interface mapStateToPropsInterface {
    isLoading:boolean;
}
const mapStateToProps = createStructuredSelector<IRootState, mapStateToPropsInterface>({
    isLoading: selectIsCollectionFetching
});

const connector = connect(
    mapStateToProps
);

type PropsFromRedux = ConnectedProps<typeof connector>;

type CollectionsOverviewContainerProps = PropsFromRedux & WithSpinnerProps;

const CollectionsOverviewContainer = compose(
    connector,
    WithSpinner
)(CollectionsOverview) as React.FC<CollectionsOverviewContainerProps>;

export default CollectionsOverviewContainer;