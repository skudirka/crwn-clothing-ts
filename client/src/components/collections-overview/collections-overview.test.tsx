import {shallow, mount, render} from 'enzyme';
import React from 'react';

import {CollectionsOverview} from './collections-overview.component';

describe('CollectionsOverview component', () => {
    let wrapper;
    const mockCollections = [
        {
            id: '1',
            name: 'Collection1',
            items: []
        },
        {
            id: '2',
            name: 'Collection2',
            items: []
        },
        {
            id: '3',
            name: 'Collection3',
            items: []
        }
    ];

    beforeEach(() => {
        const mockProps = {
            collections: mockCollections
        }
        wrapper = shallow(<CollectionsOverview {...mockProps} />);
    });

    it('should render the CollectionsOverview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a CollectionPreview for each collection', () => {
        expect(wrapper.find('CollectionPreview').length).toEqual(mockCollections.length);
    });
});