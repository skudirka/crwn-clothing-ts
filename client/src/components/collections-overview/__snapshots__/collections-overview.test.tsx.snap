// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsOverview component should render the CollectionsOverview component 1`] = `ShallowWrapper {}`;
